.github/workflows/pr-build.yaml

```diff
@@ -0,0 +1,119 @@
+name: PR build
+
+on:
+  pull_request_target:
+    types: [ opened, synchronize, reopened ]
+    paths-ignore:
+      - "LICENSE"
+      - "CODEOWNERS"
+    branches:
+      - "main"
+
+jobs:
+  cancel-previous-workflows:
+    name: Cancel previous workflows
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+    # https://github.com/styfle/cancel-workflow-action#advanced-token-permissions
+    permissions:
+      actions: write
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.0
+        with:
+          access_token: ${{ github.token }}
+
+  build:
+    name: Build application
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+    permissions:
+      contents: read
+    env:
+      DOCKER_TAG: PR-${{ github.event.pull_request.number }}
+      DOCKER_REPOSITORY: "ghcr.io/capactio/pr"
+      APP_NAME: "dashboard"
```

Comment:
You can export that to workflow envs as it's used between jobs 👍 

tested here: https://github.com/mszostok/dashboard/pull/1